lib/inferno/test_runner.rb

```diff
@@ -166,22 +156,60 @@ def update_parent_result(parent)
 
       children = parent.children(test_session.suite_options)
       child_results = results_repo.current_results_for_test_session_and_runnables(test_session.id, children)
-      required_children = children.select(&:required?)
-      required_results = child_results.select(&:required?)
-      return if required_children.length != required_results.length
+      return unless need_to_update_parent_result?(children, child_results, &parent.block)
 
+      parent_instance = parent.new
+      parent_instance.results << child_results
```

Comment:
Is this actually tested? This should create a nested array, in contrast to the flat array created above.

Reply:
`ResultCollection#<<` flattens the `results` array.